### PR TITLE
Add draft of Dask to MPI experiment

### PR DESCRIPTION
**Pull request by mrocklin**

### CODE REVIEW

File: _posts/2019-01-17-dask-mpi-experiment.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+key_to_part_dict = {str(part.key): part for part in futures_of(x)}
+who_has = client.who_has(x)
+worker_map = defaultdict(list)
+for key, workers in who_has.items():
+    worker_map[first(workers)].append(key_to_part_dict[key])
+```
+
+Admittedly, this code is gross, and not particularly friendly or obvious to
+non-Dask experts (or even Dask experts themselves, I had to steal this from the
+[Dask XGBoost project](https://dask-mpi.readthedocs.io/en/latest/), which does
```
Comment (TomAugspurger) [inline _posts/2019-01-17-dask-mpi-experiment.md]: Wrong link. for dask-xgboost

File: _posts/2019-01-17-dask-mpi-experiment.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+```
+mpirun -np 5 python my_dask_script.py
+```
+
+What just happened
+------------------
+
+So MPI started up and ran our script.  The
+[dask-mpi](https://dask-mpi.readthedocs.io/en/latest/) project set a Dask
+scheduler on rank 0, a bunch of workers on ranks 2+, and gave control back to
```
Comment (TomAugspurger) [inline _posts/2019-01-17-dask-mpi-experiment.md]: ranks 1+, instead of 2? (I've never used MPI).

File: _posts/2019-01-17-dask-mpi-experiment.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+non-Dask experts (or even Dask experts themselves, I had to steal this from the
+[Dask XGBoost project](https://dask-mpi.readthedocs.io/en/latest/), which does
+the same trick).
+
+But after that we just call our MPI library's initialize function,
+`print_data_and_rank` on all of our data using Dask's
+[Futures interface](http://docs.dask.org/en/latest/futures.html).
+That function gets the data directly from local memory (the Dask workers and
+MPI ranks are in the same process), and does whatever the MPI application
+wnats.
```
Comment (jakirkham) [inline _posts/2019-01-17-dask-mpi-experiment.md]: “wnats” -> “wants”

**Comment by mrocklin** [comment]
Thanks for the review @TomAugspurger  and @jakirkham .  I think I've resolved those issues.

**Pull request merged**